pkg/operator/controller/ingress/deployment.go

```diff
@@ -157,7 +157,7 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 	volumes := deployment.Spec.Template.Spec.Volumes
 	routerVolumeMounts := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
 
-	var desiredReplicas int32 = 2
+	var desiredReplicas int32 = 3
```

Comment:
`ensureDefaultIngressController` specifies `spec.replicas` when it creates the default ingresscontroller, so changing the default here is going to be ineffective as far as the default ingresscontroller is concerned.

`ensureDefaultIngressController` really choose a value for `spec.replicas` dynamically. The documented minimum number of worker nodes is 2, so we cannot necessarily go above that, and on clusters with more than 3 workers, it might be reasonable to scale up to, say, max(min(number of worker nodes, 5), floor(20% of worker nodes)).